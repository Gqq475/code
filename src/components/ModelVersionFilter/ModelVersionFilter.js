import React from 'react'
import classes from './ModelVersionFilter.scss'
import ModelOption from 'components/ModelOption'
import { Button } from '@blueprintjs/core'

type Props = {
getModelList: Function
};

export class ModelVersionFilter extends React.Component {
  props: Props

  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
    this.state = {
      // start: new Date(2017, 0, 1),
      // end: new Date(),
      product: '',
      operation: '',
      lineID: '',
      subEntity: '',
      pageNo: 1,
      // // Must set 1 to get total count
      pageSize: 1
      // sortBy: 'id',
      // order: 'asc'
    }
  }

  handleClick () {
    console.log(this.state)
    this.props.getModelList(this.state)
  }

  componentDidMount () {
    this.props.getModelList(this.state)
  }

  render () {
    return (
      <div className={classes['ModelVersionFilter-container']}>
        <ModelOption {...this.state} that={this} />
        <Button className={classes['btn-submit']} onClick={this.handleClick}>
          查询
        </Button>
      </div>
    )
  }
}

export default ModelVersionFilter
