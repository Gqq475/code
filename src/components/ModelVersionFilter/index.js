import ModelVersionFilter from './ModelVersionFilter'
export default ModelVersionFilter
