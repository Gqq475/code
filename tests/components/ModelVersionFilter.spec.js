import React from 'react'
import ModelVersionFilter from 'components/ModelVersionFilter/ModelVersionFilter'

describe('(Component) ModelVersionFilter', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <ModelVersionFilter />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
