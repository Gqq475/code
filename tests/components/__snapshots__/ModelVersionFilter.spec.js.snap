// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ModelVersionFilter render normally 1`] = `
<div
  className="ModelVersionFilter-container"
>
  <ModelOption
    lineID=""
    operation=""
    pageNo={1}
    pageSize={1}
    product=""
    subEntity=""
    that={
      ModelVersionFilter {
        "_reactInternalInstance": ShallowComponentWrapper {
          "_calledComponentWillUnmount": false,
          "_compositeType": 0,
          "_context": Object {},
          "_currentElement": <ModelVersionFilter />,
          "_debugID": 1,
          "_hostContainerInfo": null,
          "_hostParent": null,
          "_instance": [Circular],
          "_mountOrder": 1,
          "_pendingCallbacks": null,
          "_pendingElement": null,
          "_pendingForceUpdate": false,
          "_pendingReplaceState": false,
          "_pendingStateQueue": null,
          "_renderedComponent": NoopInternalComponent {
            "_currentElement": <div
              className="ModelVersionFilter-container"
      >
              <ModelOption
                      lineID=""
                      operation=""
                      pageNo={1}
                      pageSize={1}
                      product=""
                      subEntity=""
                      that={[Circular]}
              />
              <Blueprint.Button
                      className="btn-submit"
                      onClick={[Function]}
              >
                      查询
              </Blueprint.Button>
      </div>,
            "_debugID": 2,
            "_renderedOutput": <div
              className="ModelVersionFilter-container"
      >
              <ModelOption
                      lineID=""
                      operation=""
                      pageNo={1}
                      pageSize={1}
                      product=""
                      subEntity=""
                      that={[Circular]}
              />
              <Blueprint.Button
                      className="btn-submit"
                      onClick={[Function]}
              >
                      查询
              </Blueprint.Button>
      </div>,
          },
          "_renderedNodeType": 0,
          "_rootNodeID": 0,
          "_topLevelWrapper": null,
          "_updateBatchNumber": null,
          "_warnedAboutRefsInRender": false,
        },
        "context": Object {},
        "handleClick": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "lineID": "",
          "operation": "",
          "pageNo": 1,
          "pageSize": 1,
          "product": "",
          "subEntity": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      }
    }
  />
  <Blueprint.Button
    className="btn-submit"
    onClick={[Function]}
  >
    查询
  </Blueprint.Button>
</div>
`;
